src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -9723,123 +9722,115 @@ namespace SurfaceGeometry {
                                           "Valid shading control types for exterior window screens are ALWAYSON, ALWAYSOFF, or ONIFSCHEDULEALLOWS.");
                     }
                 } else if (state.dataIPShortCut->lAlphaFieldBlanks(4)) {
                     ShowSevereError(state,
-                                    cCurrentModuleObject + "=\"" + state.dataSurface->WindowShadingControl(ControlNum).Name + "\", " +
-                                        state.dataIPShortCut->cAlphaFieldNames(4) + " is blank.");
+                                    cCurrentModuleObject + "=\"" + windowShadingControl.Name + "\", " + state.dataIPShortCut->cAlphaFieldNames(4) +
+                                        " is blank.");
                     ShowContinueError(state, "A valid construction is required.");
                     ErrorsFound = true;
                 } else {
                     ShowSevereError(state,
-                                    cCurrentModuleObject + "=\"" + state.dataSurface->WindowShadingControl(ControlNum).Name + "\", " +
-                                        state.dataIPShortCut->cAlphaFieldNames(4) + " is invalid.");
+                                    cCurrentModuleObject + "=\"" + windowShadingControl.Name + "\", " + state.dataIPShortCut->cAlphaFieldNames(4) +
+                                        " is invalid.");
                     ShowContinueError(state,
                                       "Construction=\"" + state.dataIPShortCut->cAlphaArgs(4) + "\" was used. A valid construction is required.");
                     ErrorsFound = true;
                 }
             }
 
             // Warning if setpoint is unintentionally zero
-            if (state.dataSurface->WindowShadingControl(ControlNum).SetPoint == 0 && ControlType != "ALWAYSON" && ControlType != "ALWAYSOFF" &&
+            if (windowShadingControl.SetPoint == 0 && ControlType != "ALWAYSON" && ControlType != "ALWAYSOFF" &&
                 ControlType != "ONIFSCHEDULEALLOWS" && ControlType != "SCHEDULE" && ControlType != "ONIFHIGHGLARE" && ControlType != "GLARE" &&
                 ControlType != "DAYLIGHTILLUMINANCE") {
-                ShowWarningError(state,
-                                 cCurrentModuleObject + "=\"" + state.dataSurface->WindowShadingControl(ControlNum).Name +
-                                     "\", The first SetPoint is zero.");
+                ShowWarningError(state, cCurrentModuleObject + "=\"" + windowShadingControl.Name + "\", The first SetPoint is zero.");
                 ShowContinueError(state, "..You may have forgotten to specify that setpoint.");
             }
 
             // Upward compatibility for old Shading Control Type names
             if (ControlType == "SOLARONWINDOW" || ControlType == "HORIZONTALSOLAR" || ControlType == "OUTSIDEAIRTEMP" ||
                 ControlType == "ZONEAIRTEMP" || ControlType == "ZONECOOLING") {
                 ControlType = "ONIFHIGH" + ControlType;
```

Comment:
Need to find a `getEnumerationValue` type solution for this. I suggest multiple `std::array<std::string_view, static_cast<int>(X::Num)> XNamesUC = {}, XNamesUC1 = {};` and then `getEnumerationValue` becomes a template or we have multiple calls to it where we need them.

Reply:
While we are at this, move `getEnumerationValue` into a `.cc` file and make that function non `constexpr`.

Reply:
There's not need to support old type names. They won't get past the input processor now, they will fail on the key choice validation. Here's the valid choices from the idd:
```
  A5 , \field Shading Control Type
       \required-field
       \type choice
       \key AlwaysOn
       \key AlwaysOff
       \key OnIfScheduleAllows
       \key OnIfHighSolarOnWindow
       \key OnIfHighHorizontalSolar
       \key OnIfHighOutdoorAirTemperature
       \key OnIfHighZoneAirTemperature
       \key OnIfHighZoneCooling
       \key OnIfHighGlare
       \key MeetDaylightIlluminanceSetpoint
       \key OnNightIfLowOutdoorTempAndOffDay
       \key OnNightIfLowInsideTempAndOffDay
       \key OnNightIfHeatingAndOffDay
       \key OnNightIfLowOutdoorTempAndOnDayIfCooling
       \key OnNightIfHeatingAndOnDayIfCooling
       \key OffNightAndOnDayIfCoolingAndHighSolarOnWindow
       \key OnNightAndOnDayIfCoolingAndHighSolarOnWindow
       \key OnIfHighOutdoorAirTempAndHighSolarOnWindow
       \key OnIfHighOutdoorAirTempAndHighHorizontalSolar
       \key OnIfHighZoneAirTempAndHighSolarOnWindow
       \key OnIfHighZoneAirTempAndHighHorizontalSolar
```

Reply:
Hallelujah!